scripts/project.js

```diff
@@ -1,27 +1,51 @@
-var projects = [];
-
-function Project(opts) {
+function Projects(opts) {
   for (key in opts) this[key] = opts[key];
 }
 
-Project.prototype.toHtml = function() {
-  var source = $('#projectData-template').html();
-  var template = Handlebars.compile(source);
+Projects.all = [];
+
+Projects.prototype.toHtml = function() {
+  // var source = $('#projectData-template').html();
+  var template = Handlebars.compile($('#projectData-template').text());
   this.daysAgo = parseInt((new Date() - new Date(this.publishedOn))/60/60/24/1000);
   this.publishStatus = this.publishedOn ? 'published ' + this.daysAgo + ' days ago' : '(draft)';
+  // this.body = marked(this.body); WHY DOESN'T THIS WORK?
   return template(this);
 };
 
-projectData.sort(function(a,b){
-  return (new Date(b.publishedOn)) - (new Date(a.publishedOn));
-});
-
-projectData.forEach(function(ele) {
-  projects.push(new Project(ele));
-});
-
-projects.forEach(function(a){
-  $('#portfolio').append(a.toHtml());
-});
+Projects.loadAll = function(rawData) {
+  rawData.sort(function(a,b){
+    return (new Date(b.publishedOn)) - (new Date(a.publishedOn));
+  });
+  rawData.forEach(function(ele) {
+    Projects.all.push(new Projects(ele));
+  });
+};
+// projects.forEach(function(a){
+//   $('#portfolio').append(a.toHtml());
+// });
+//
+// $('.template').hide();
 
-$('.template').hide();
+Projects.fetchAll = function() {
+  var localEtag;
+  $.ajax({
+    type: 'HEAD',
+    url: 'scripts/projectData.json',
+    success: function(data, message, xhr) {
+      localEtag = xhr.getResponseHeader('eTag');
+      localStorage.setItem('eTag', localEtag);
+      if (localStorage.eTag === xhr.getResponseHeader('eTag') && localStorage.rawData) {
```

Comment:
You have a slight issue here.  You should step through this and see how it is working.  It looks like you are setting the etag before you check what it is.  You should be checking to see if the localStorage.etag is the same as the one you are getting back from the response first.  To see if it is the same. Then proceeding from there based on wether it is the same or not.